okcupyd/json_search.py

```diff
@@ -118,17 +167,539 @@ def __call__(self, response_dictionary):
+                raise TypeError
+        return education
+
+
+class ChildrenFilter(search_filters.filter_class):
+    output_key = "children"
+    decide = search_filters.any_not_none_decider
+    acceptable_values = ((True, 'has a kid', 'has kids', False, "doesn't have a kid", "doesn't have kids"),
+                         (True, 'wants', 'wants kids', False, "doesn't want", "doesn't want kids",
+                         'might want', 'might want kids'))
+    # types = (str, str)
+    # TODO: also permit single key `kids`?
+    def transform(has_kids=None, wants_kids=None):
+        children = []
+        if isinstance(has_kids, six.string_types):
+            has_kids = has_kids.lower()
+        if isinstance(wants_kids, six.string_types):
+            wants_kids = wants_kids.lower()
+        if has_kids == 'has a kid' or has_kids == 'has kids' or has_kids is True:
+            children.append('has_one_or_more')
+        elif has_kids == "doesn't have a kid" or has_kids == "doesn't have kids" or has_kids is False:
+            children.append('doesnt_have')
+        elif has_kids is not None:
+            raise TypeError
+        # else: children.extend(('has_one_or_more', 'doesnt_have'))
+        if wants_kids == 'wants' or wants_kids == 'wants kids':
+            children.append('wants_kids')
+        elif wants_kids == "doesn't want" or wants_kids == "doesn't want kids":
+            children.append('doesnt_want')
+        elif wants_kids == 'might want' or wants_kids == 'might want kids':
+            children.append('might_want')
+        elif wants_kids is True:
+            children.extend(('wants_kids', 'might_want'))
+        elif wants_kids is False:
+            children.extend(('doesnt_want', 'might_want'))
+        elif wants_kids is not None:
+            raise TypeError
+        # else: children.extend(('wants_kids', 'might_want', 'doesnt_want'))
+        return children
+
+
+def normalform(keys, magicmap):
+    result = []
+    for k in keys:
+        i = magicmap[k]
+        result.append(magicmap.values[i])
+    return result
+
+
+class EthnicitiesFilter(search_filters.filter_class):
+    output_key = "ethnicity"
+    # description = "One or more ethnicities that search results must match."
+    acceptable_values = (
+        'asian',  'black', 'hispanic', 'hispanic/latin', 'indian', 'latin',
+        'middle eastern', 'native american', 'pacific islander',
+        'white', 'other')
+    types = "string or list of strings"
+    @util.makelist_decorator
+    def transform(ethnicities=[]):
+        return normalform(ethnicities, magicnumbers.maps.ethnicities)
+
+
+class ReligionFilter(search_filters.filter_class):
+    # output_key = "religion"
+    # description = "One or more religions that search results must match."
+    acceptable_values = magicnumbers.maps.religion.keys()
+    types = "string or list of strings"
+    @util.makelist_decorator
+    def transform(religion=[]):
```

Comment:
I know the issue. As I understand the current architecture, these default values will never be used; explicit arguments will always be supplied to the `transform` function. The default value were just my way for the moment of documenting what defaults should go into the request dict. Not saying they're a very good way of documenting that.